packages/runtime/test-runtime-utils/src/mockLogger.ts

```diff
@@ -21,13 +21,34 @@ export class MockLogger extends TelemetryLogger implements ITelemetryLogger {
     }
 
     /**
-     * Search events logged since the last time matchEvents was called,
-     * looking for the given expected events in order.
+     * Search events logged since the last time matchEvents was called, looking for the given expected
+     * events in order.
      * @param expectedEvents - events in order that are expected to appear in the recorded log.
      * These event objects may be subsets of the logged events.
      * Note: category is ommitted from the type because it's usually uninteresting and tedious to type.
      */
     matchEvents(expectedEvents: Omit<ITelemetryBaseEvent, "category">[]): boolean {
+        const matchedExpectedEventCount = this.getMatchedEventsCount(expectedEvents);
+        // How many expected events were left over? Hopefully none.
+        const unmatchedExpectedEventCount = expectedEvents.length - matchedExpectedEventCount;
+        assert(unmatchedExpectedEventCount >= 0);
```

Comment:
Is this assert needed here? Shouldn't the caller assert on the return value of this function?

Reply:
I just moved the code around but yeah, you are right. We should not assert here. Removed it.